kushaldas/sdstatus
# Rewrite it in Rust (RIIR)

Tinkering around with rust & tokio, I figured the `sdstatus` tool would be simple to rewrite given that we only use it internally. Questions to ask yourself:

* Do we even want this in rust? It takes a while to compile.
* Did I port all the useful options? I made sure `scan` and `l10n` work, but I didn't bother to re-implement `--format=csv`, everything is JSON now. 
* Does it work for you?

If all that's true, then please take a look at the code and modify to your heart's content. Also, I would not in the least be offended if this PR gets closed as too confusing—mostly sharing it as a practical example of "hey, prototyping in rust can be fun, let's do more of this!" :innocent: 

To test:

1. Make sure you've got rust installed locally (see https://rustup.rs/ if not)
2. `cargo build`
3. `cargo run -- scan` should use directory info by default
4. `cargo run -- scan <onion url>` should scan just the 1 onion
5. `cargo run -- l10n <(cargo run -- scan)` should spit out a localization report
